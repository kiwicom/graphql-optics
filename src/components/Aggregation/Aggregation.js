import React from 'react'
import { withRouter } from 'react-router-dom'
import Button from '../Button/Button'
import vars from '../../styles/vars'

const style = {
  fontFamily: vars.fontFamily,
}

const Aggregation = ({ aggregation, history, isActive }) => (
  <Button
    style={style}
    isActive={isActive}
    onClick={() => history.push(`/aggregations/${aggregation.key}`)}
  >
    {aggregation.key} - {aggregation.doc_count}
  </Button>
)

export default withRouter(Aggregation)
