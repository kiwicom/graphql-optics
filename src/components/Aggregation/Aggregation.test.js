import React from 'react'
import Aggregation from './Aggregation'
import { renderWithRouter } from '../../__tests__/helpers'

it('renders correctly', () => {
  const data = {
    key: 'allLocations',
    doc_count: 6,
  }
  const tree = renderWithRouter({
    children: <Aggregation aggregation={data} />,
  })
  expect(tree).toMatchSnapshot()
})
