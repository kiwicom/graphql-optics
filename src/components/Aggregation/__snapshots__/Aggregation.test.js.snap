// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="button"
  onClick={[Function]}
  style={
    Object {
      "fontFamily": "sans-serif",
    }
  }
>
  allLocations
   - 
  6
</div>
`;
